infrastructure/terraform/modules/gcp-gke/main.tf

```diff
@@ -0,0 +1,76 @@
+provider "google" {
+  project = var.project_id
+  region  = var.region
+}
+
+# VPC
+resource "google_compute_network" "vpc" {
+  name                    = var.vpc_name
+  auto_create_subnetworks = "false"
+}
+
+# Subnet # private_ip_google_access = false by default
+resource "google_compute_subnetwork" "subnet" {
+  name          = "${var.project_id}-subnet-public-0"
+  region        = var.region
+  network       = google_compute_network.vpc.name
+  ip_cidr_range = var.public_subnets[0]
+}
+
+# GKE cluster
+resource "google_container_cluster" "primary" {
+  name     = "${var.project_id}-gke"
+  location = var.zone
+  
+  # We can't create a cluster with no node pool defined, but we want to only use
+  # separately managed node pools. So, we create the smallest possible default
+  # node pool and immediately delete it.
```

Comment:
Also these comments we can remove I think.

Reply:
👍🏼